src/odb/src/db/dbSite.cpp

```diff
@@ -230,6 +232,62 @@ void dbSite::setClass(dbSiteClass type)
   site->_flags._class = type.getValue();
 }
 
+void dbSite::setRowPattern(
+    std::vector<std::pair<std::string, std::string>>& row_pattern)
+{
+  _dbSite* site = (_dbSite*) this;
+  site->_row_patterns.reserve(row_pattern.size());
+  site->_flags._is_hybrid = 1;
+  std::unordered_map<std::string, dbOrientType::Value> orientationMap
+      = {{"N", dbOrientType::R0},
+         {"W", dbOrientType::R270},
+         {"S", dbOrientType::R180},
+         {"E", dbOrientType::R90},
+         {"FN", dbOrientType::MYR90},
+         {"FW", dbOrientType::MY},
+         {"FS", dbOrientType::MXR90},
+         {"FE", dbOrientType::MX}};
+
+  for (auto& row : row_pattern) {
+    dbOrientType orient = dbOrientType::R0;
+    auto _site = (_dbSite*) getLib()->findSite(row.first.c_str());
+    _site->_flags._is_hybrid = 1;
+    auto it = orientationMap.find(row.second);
+    if (it != orientationMap.end()) {
+      orient = it->second;
+    }
+    site->_row_patterns.emplace_back(row.first, orient);
+  }
+}
+
+bool dbSite::hasRowPattern()
+{
+  _dbSite* site = (_dbSite*) this;
+  return !site->_row_patterns.empty();
+}
+
+bool dbSite::isHybrid()
```

Comment:
What is the difference between isHybrid & hasRowPattern?  Do we need both (same about _row_pattern & _is_hybird)?

Reply:
Yeah there is a big difference.
in the case of having a hybridCell AB that is made of A&B. All 3 are hybrid cells, but only cell AB would have a row pattern set of <A,orientation> and <B,orientation>